auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java

```diff
@@ -57,7 +57,10 @@
  *
  * <p>Uses the new and simplified Google auth library:
  * https://github.com/google/google-auth-library-java
+ *
+ * @deprecated use {@link GoogleAuthLibraryCallCredentials} instead.
  */
+@Deprecated
```

Comment:
Should we replace the code in this class with `next.newCall(method, callOptions.withCredentials(googleCredentials))`?